### PR TITLE
modify gogo to generate (more) stable import names

### PR DESCRIPTION
**Pull request by PatrickDropbox**
This reduces the churn in the generated code (at dropbox, we check in the generated code into our repos)

### CODE REVIEW

**Comment by awalterschulze** [comment]
We also check in generated code into our repos.

If I run make all
> panic: assignment to entry in nil map
Please run at least make all and make testall before submitting a pull request.

Maybe you want to explain how this reduces churn?

**Comment by PatrickDropbox** [comment]
ah, much better (I only ran make tests before)

take a look at the test files.  we were importing the same packages multiple times with different name suffixes; these suffixes can change whenever the proto definitions are modified.  This results in a massive diffs whenever we recompile and check in proto files

File: protoc-gen-gogo/generator/helper.go [hunk 1]
```diff
@@ -99,7 +99,10 @@ func (this *pluginImports) GenerateImports(file *FileDescriptor) {
 type Single interface {
 	Use() string
 	IsUsed() bool
-	Generate() string
+	Name() string
+	Location() string
+
+	Generate() string // DEPRECATED
```
Comment (awalterschulze) [inline protoc-gen-gogo/generator/helper.go]: Please delete this method in the interface and in the implementation.
I don't like deprecated code lying around its not a necessity.

File: test/thetest.proto [hunk 2]
```diff
@@ -583,10 +583,10 @@ message UnrecognizedWithEmbed {
 }
 
 //begin versioned: only for protoc 2.6.0 and up
-message SampleOneOf {
-  oneof test_oneof {
-    string name = 4;
-    NinOptNative sub_message = 9;
-  }
-}
+// // message SampleOneOf {
```
Comment (awalterschulze) [inline test/thetest.proto]: I have made a fix for this that I haven't pushed yet.
My bad.
If possible could you please revert the change on this file.
Sorry.

**Comment by awalterschulze** [comment]
This is a great patch.
Thank you very much.

**Comment by PatrickDropbox** [comment]
rm deprecated function + reverted thetests.proto.

**Comment by awalterschulze** [comment]
Thank you very much this is great :)
